lib/builtin.t

```diff
@@ -49,6 +49,22 @@ package FetchStoreCounter {
     is($storecount, 1, 'isbool() TARG invokes STORE magic');
 }
 
+# weakrefs
+{
+    use builtin qw( isweak weaken unweaken );
+
+    my $arr = [];
+    my $ref = $arr;
+
+    ok(!isweak($ref), 'ref is not weak initially');
+
+    weaken($ref);
+    ok(isweak($ref), 'ref is weak after weaken()');
+
+    unweaken($ref);
+    ok(!isweak($ref), 'ref is not weak after unweaken()');
+}
```

Comment:
How about a test that `undef`ing the strong ref also `undef`s the weak one, i.e. that it has the desired practical effect, not just satisfying the predicate function.

Reply:
Yeah, I guess. I didn't want to go too far into testing the concept of weakrefs here because I'm sure that'll exist elsewhere, but I guess we could test it enough to demonstrate we've in fact connected the builtin functions to the right core concept. Done in dfa4906e94